Sources/Host/POSIX/File.cpp

```diff
@@ -118,6 +128,40 @@ ErrorCode File::pwrite(ByteVector const &buf, uint64_t &count,
   return _lastError = kSuccess;
 }
 
+// lldb expects stat data is returned as a packed buffer with total size of 64
+// bytes. The field order is the same as the POSIX defined stat struct. All
+// fields are encoded as 4-byte, big-endian unsigned integers except for
+// st_size, st_blksize, and st_blocks which are all 8-byte, big-endian unsigned
+// integers.
+ErrorCode File::fstat(ByteVector &buffer) const {
+  struct stat s;
+  if (::fstat(_fd, &s) < 0)
+    return Platform::TranslateError();
+
+  const auto appendInteger = [&buffer](auto value) -> void {
+    buffer.insert(buffer.end(),
```

Comment:
Might be nice to add a `DS2ASSERT(sizeof(value) == 16)`.

Reply:
`sizeof(value)` is either 8 or 4 here so I can add an assert for that, but did you intend something else?

Reply:
Err, sorry, I meant, 2 not 16. We are only appending 2 bytes.

Reply:
It always appends either 4 or 8 bytes with the current usage.

Reply:
Maybe we're talking about different things?

Here, `value` is always explicitly cast to `uint64_t` or `uint32_t` by the caller, and `sizeof()` on the `auto` typed `value` argument resolves to either 8 or 4, respectively.

I could extract `appendInteger `to a separate method with a template type for `value` instead of `auto` if that would make it clearer-- a local function was a bit more brief, but can't be a template (which I think is more clear than `auto`).

Reply:
Sure, it is only ever passed a `uint64_t` or `uint32_t`, but, the buffer just does a cast to `uint8_t` and uses [0] and [1], which would be the first two bytes right? We _could_ consider bumping to C++20 which would give us templated lambdas 🤔

Reply:
The `&value + 1` is pre-cast to `uint8_t*`, so it will capture either a 4 -or 8-byte range. But how about I rewrite it as follows to make it more clear?
```
    buffer.insert(buffer.end(),
                  reinterpret_cast<uint8_t*>(&value),
                  reinterpret_cast<uint8_t*>(&value) + sizeof(value));
```

Reply:
Oh! I see, yeah ... that is definitely a bit easier to read.